### PR TITLE
Use new ic-management library

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Introduces `@dfinity/ic-management` while minimizing the changes.

# Changes

- add `@dfinity/ic-management` and bump `next` libs
- intregrate `ICManagementCanister` from ic-js within `ICManagementCanister` of NNS dapp

### CODE REVIEW

File: frontend/src/lib/canisters/ic-management/converters.ts [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 import type { CanisterDetails } from "./ic-management.canister.types";
 import { CanisterStatus } from "./ic-management.canister.types";
-import type {
-  CanisterStatusResponse,
-  CanisterStatus as RawCanisterStatus,
-} from "./ic-management.types";
 
-const getCanisterStatus = (status: RawCanisterStatus): CanisterStatus => {
+const getCanisterStatus = (
+  status:
```
Comment (lmuntaner) [inline frontend/src/lib/canisters/ic-management/converters.ts:7]: Why aren't you using `CanisterStatusResponse`?

File: frontend/src/lib/canisters/ic-management/ic-management.canister.ts
Comment (lmuntaner) [inline frontend/src/lib/canisters/ic-management/ic-management.canister.ts:1]: Will you then remove this and create the canister directly in the api layer like we do in the other parts?

The transformations can happen at that level as well.

**Review by lmuntaner — APPROVED** [review]
LGTM, just a couple of questions.

File: frontend/src/lib/canisters/ic-management/converters.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/canisters/ic-management/converters.ts:7]: Where? For what?

File: frontend/src/lib/canisters/ic-management/ic-management.canister.ts
Comment (peterpeterparker) [reply frontend/src/lib/canisters/ic-management/ic-management.canister.ts:1]: Ultimately yes agree I think we should do it. I added a `TODO` for that but, I might priorize other tasks right now.

Comment (lmuntaner) [reply frontend/src/lib/canisters/ic-management/ic-management.canister.ts:1]: Ok, thanks

File: frontend/src/lib/canisters/ic-management/converters.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/canisters/ic-management/converters.ts:7]: Don't we have this type in the library:

```
{ stopped: null } | { stopping: null } | { running: null }
```

Comment (peterpeterparker) [reply frontend/src/lib/canisters/ic-management/converters.ts:7]: Ah no, didc generates `{ stopped: null } | { stopping: null } | { running: null }` and no type for it. That was a custom type.

Comment (lmuntaner) [reply frontend/src/lib/canisters/ic-management/converters.ts:7]: Ah, ok, I see now it's part of the whole response, not a specific one. Thanks for the explanation!

**Pull request merged**